test/e2e_node/cpumanager_test.go

```diff
@@ -263,20 +278,21 @@ var _ = SIGDescribe("CPU Manager", ginkgo.Ordered, framework.WithSerial(), featu
 	})
 
 	ginkgo.When("running guaranteed pod tests", ginkgo.Label("guaranteed", "exclusive-cpus"), func() {
-		var cpuAlloc int64
 		var reservedCPUs cpuset.CPUSet
 
 		ginkgo.BeforeEach(func(ctx context.Context) {
 			reservedCPUs = cpuset.New(0)
-
-			localNode = getLocalNode(ctx, f)
-			cpuAllocQty := localNode.Status.Allocatable[v1.ResourceCPU]
-			cpuAlloc = cpuAllocQty.Value()
 		})
 
 		ginkgo.It("should allocate exclusively a CPU to a 1-container pod", func(ctx context.Context) {
 			cpuCount := 1
 
+			cpuReq := int64(cpuCount + reservedCPUs.Size())
+			cpuDetails = cpuDetailsFromNode(getLocalNode(ctx, f)) // avoid potentially costly updateKubeletConfigIfNeeded
+			if cpuDetails.Allocatable < cpuReq {
+				e2eskipper.Skipf("Skipping CPU Manager tests since the CPU allocatable %d < CPU request %d (online %d)", cpuDetails.Allocatable, cpuReq, onlineCPUs.Size())
+			}
```

Comment:
This seems to appear repeatedly, can we try to abstract into a helper like below?:

func skipIfInsufficientCPU(ctx context.Context, f *framework.Framework, cpuReq int64, onlineCPUs cpuset.CPUSet) bool

Reply:
ok since there is frequent request from reviewers, I'll extract a helper function